### ISSUE TITLE
Deserializing a HashMap field fails

### ISSUE THREAD

**Issue by dnaka91** [samscott89/serde_qs#49]
I tried using a `HashMap` as field and serializing works just fine. When trying to deserialize the value back I get an error though.

The struct is as simple as:

```rust
struct Mapping {
    mapping: HashMap<u64, u64>,
}
```

And serializes into the following as expected:
```txt
mapping[1]=2&mapping[3]=4
```

When deserializing back into the same struct I get the error:
```txt
failed with reason: invalid type: sequence, expected a map
```

I fortunately got around it by directly using the `HashMap` without the struct, as I currently have only that single field. Still, it would be great to be able to deserialize the data that can be produced by the crate (and because I'll likely have more than 1 field in the future).

**Comment by samscott89** [comment]
Hey @dnaka91 . Please can you try the latest changes merged to main and let me know if it works. I'll publish a new release with the fix once you've confirmed.

**Comment by dnaka91** [comment]
Sorry for the late reply here as well.

I could confirm that it works with the latest changes from the main branch. Thank you so much for the quick fix :+1:

**Comment by samscott89** [comment]
> 
> 
> Sorry for the late reply here as well.
> 
> I could confirm that it works with the latest changes from the main branch. Thank you so much for the quick fix 👍

Thanks for raising this, it's a good addition!

**Comment by samscott89** [comment]
Published in 0.8.4
